### PR TITLE
Annotate default groups in conflict error messages

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

We now tell the user if a group was enabled by default.

Closes https://github.com/astral-sh/uv/issues/9366.

### CODE REVIEW

File: crates/uv/tests/it/lock.rs [hunk 1]
```diff
@@ -3765,6 +3765,197 @@ fn lock_conflicting_group_basic() -> Result<()> {
...
+        name = "project"
+        version = "0.1.0"
+        description = "Add your description here"
+        requires-python = ">=3.12"
+
+        [tool.uv]
+        conflicts = [
+            [
+              { group = "project1" },
+              { group = "project2" },
```
Comment (zanieb) [inline crates/uv/tests/it/lock.rs:3789]: As in the other pull request, can we name these differently please?

**Review by zanieb — APPROVED** [review]
Love this :)

File: crates/uv/src/commands/project/mod.rs [hunk 2]
```diff
@@ -227,6 +209,43 @@ pub(crate) enum ProjectError {
...
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{} are incompatible with the declared conflicts: {{{}}}",
+            self.conflicts
+                .iter()
+                .map(|conflict| match conflict {
+                    ConflictPackage::Extra(ref extra) => format!("extra `{extra}`"),
+                    ConflictPackage::Group(ref group) if self.dev.default(group) =>
+                        format!("group `{group}` (enabled by default)"),
```
Comment (zanieb) [inline crates/uv/src/commands/project/mod.rs:232]: Another option is just "default group `{group}`"

File: crates/uv/tests/it/lock.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv/tests/it/lock.rs:3789]: This series of tests all use this style, I guess I'd rather do it in a single PR and change all of them.

Comment (zanieb) [reply crates/uv/tests/it/lock.rs:3789]: Yeah no problem.

**Pull request merged**